src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -1255,12 +1253,47 @@ private void validateStaticCodeAnalysisSettings(ProgrammingExercise programmingE
         return ResponseEntity.ok().build();
     }
 
-    private void validateAndAddAuxiliaryRepositoriesOfProgrammingExercise(ProgrammingExercise programmingExercise, List<AuxiliaryRepository> newAuxiliaryRepositories,
-            ProgrammingExercise updatedExercise) {
-        final List<AuxiliaryRepository> auxiliaryRepositories = Objects.requireNonNullElse(programmingExercise.getAuxiliaryRepositories(), new ArrayList<AuxiliaryRepository>())
-                .stream().filter(repo -> repo.getId() != null).collect(Collectors.toList());
+    private void validateAndAddAuxiliaryRepositoriesOfProgrammingExercise(ProgrammingExercise programmingExercise, List<AuxiliaryRepository> newAuxiliaryRepositories) {
```

Comment:
I am not sure if this resource is the right place for the validation checks. Since they appear to be only dependent on their parameters, moving them to a utilities class might help reduce the size of this file